### PR TITLE
Add unit tests for `Checklist`

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Convert `validate_options_dict` and `validate_value` to re-usable validators
- Indicate `_set_actions` validator as private in all selectors
- Provide defaults to options and value in checklist directly
- Add unit tests for checklist

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [x] I have added the PR number to the change description in the changelog fragment, e.g. `Enable feature XXX (#123)` (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: vizro-core/src/vizro/models/_components/form/_form_utils.py [hunk 1]
```diff
@@ -17,3 +19,40 @@ def get_options_and_default(options: OptionsType, multi: bool = False):
         default_value = options[0]
 
     return options, default_value
+
+
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/_form_utils.py:23]: Maybe just to add a comment "# Validators utils"

File: vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        checklist = Checklist(options=options, value=test_value)
+
+        assert hasattr(checklist, "id")
+        assert checklist.type == "checklist"
+        assert checklist.value == test_value
+        assert checklist.title is None
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize(
+        "test_value, options",
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py:117]: Can we preserve parameter order in those two tests (or

"options, test_value", or
"test_value, options",
?

File: vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+class TestChecklistBuild:
+    """Tests model build method."""
+
+    def test_checklist_build(self, expected_checklist):
+        checklist = Checklist(options=["A", "B", "C"], id="checklist_id", title="Title").build()
+
+        result = json.loads(json.dumps(checklist, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_checklist, cls=plotly.utils.PlotlyJSONEncoder))
+        assert result == expected
```
Comment (petar-qb) [inline vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py:144]: The fixture `expected_checklist` contains the option "ALL". Where does it come from if there is no multi=True set?

**Review by petar-qb — COMMENTED** [review]
Looks good! I'm only interested in one thing (test `test_checklist_build` I commented below).

File: vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py (hunk 3 above)
Comment (huong-li-nguyen) [reply vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py:144]: The dcc.Checklist is by default a multi-select (so in that sense multi=True is always true). There is no multi option though. Similar to the dcc.RadioItem being a single-select by default. It's only the Dropdown where you can select between multi or single-select

Comment (petar-qb) [reply vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py:144]: Yupp, I see now. Thanks.

**Review by petar-qb — APPROVED** [review]

**Review by nadijagraca — APPROVED** [review]
Great job Li, I love all the re-usable validators you added.

**Pull request merged**